test/distributed/elastic/rendezvous/utils_test.py

```diff
@@ -215,3 +220,103 @@ def test_matches_machine_hostname_returns_false_if_hostname_does_not_match(self)
         for host in hosts:
             with self.subTest(host=host):
                 self.assertFalse(_matches_machine_hostname(host))
+
+
+class PeriodicTimerTest(TestCase):
+    def test_start_can_be_called_only_once(self):
+        timer = _PeriodicTimer(timedelta(seconds=1), lambda: None)
+
+        timer.start()
+
+        with self.assertRaisesRegex(RuntimeError, r"^The timer has already started.$"):
+            timer.start()
+
+        timer.cancel()
+
+    def test_cancel_can_be_called_multiple_times(self):
+        timer = _PeriodicTimer(timedelta(seconds=1), lambda: None)
+
+        timer.start()
+
+        timer.cancel()
+        timer.cancel()
+
+    def test_cancel_stops_background_thread(self):
+        timer = _PeriodicTimer(timedelta(seconds=1), lambda: None)
+
+        timer.start()
+
+        self.assertTrue(any(t.name == "PeriodicTimer" for t in threading.enumerate()))
+
+        timer.cancel()
+
+        self.assertTrue(all(t.name != "PeriodicTimer" for t in threading.enumerate()))
+
+    def test_delete_stops_background_thread(self):
+        timer = _PeriodicTimer(timedelta(seconds=1), lambda: None)
+
+        timer.start()
+
+        self.assertTrue(any(t.name == "PeriodicTimer" for t in threading.enumerate()))
+
+        del timer
+
+        self.assertTrue(all(t.name != "PeriodicTimer" for t in threading.enumerate()))
+
+    def test_timer_calls_background_thread_at_regular_intervals(self):
+        begin_time = time.monotonic()
```

Comment:
nit: maybe set the begin_time right before timer.start below?

Reply:
Makes sense. Will do in the next revision.